### PR TITLE
Refactor "deployments" sql query

### PR DESCRIPTION
**Pull request by philipdalen**
- Pulled the different subqueries out into it's own WITH clauses as recommended by the BigQuery docs (please see screenshot below)
- I didn't change any of the functionality, the query still output the same result. I just found it really difficult to keep all the variables in my head because of nesting, which motivated me to refactor the query a bit.

https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax#with_clause
image

### CODE REVIEW

**Comment by philipdalen** [comment]
@dinagraves I think you were the original programmer of the query, so it's totally up to you're judgement if the query is more readable, I won't take offense it you reject the PR :)

**Review by dinagraves — APPROVED** [review]
Awesome! Thank you!  What you're looking at is someone who learned SQL on a system that didn't support `WITH` statements and so I tend to shy away from them myself! 

But I do agree, this is easier for people to digest.  

Some style nits!  Thank you again for the contribution!

File: queries/deployments.sql [hunk 1]
```diff
@@ -55,16 +39,41 @@ FROM(
...
+        source,
+        deploy_id,
+        deploys.time_created time_created,
+        change_metadata,        
+        json2array(JSON_EXTRACT(change_metadata, '$.commits')) as array_commits,
+        main_commit
+        FROM deploys 
+        JOIN
+        changes_raw on (
+        changes_raw.id = deploys.main_commit
```
[the remark this answers is not in the document]
Comment (philipdalen) [reply queries/deployments.sql]: Done.

File: queries/deployments.sql [hunk 2]
```diff
@@ -55,16 +39,41 @@ FROM(
...
+        UNION ALL 
+        SELECT * FROM deploys_tekton        
+    ),
+    changes_raw AS (
+        SELECT
+        id,
+        metadata as change_metadata
+        FROM four_keys.events_raw        
+    ),
+    changes as (
```
[the remark this answers is not in the document]
Comment (philipdalen) [reply queries/deployments.sql]: Done.

File: queries/deployments.sql [hunk 3]
```diff
@@ -55,16 +39,41 @@ FROM(
...
+        )
+    )
+
+    SELECT 
+    source,
+    deploy_id,
+    time_created,
+    main_commit,   
+    ARRAY_AGG(DISTINCT JSON_EXTRACT_SCALAR(array_commits, '$.id')) changes,    
+    FROM changes
```
Comment (dinagraves) [inline queries/deployments.sql:77]:
```suggestion
    FROM deployment_changes
```

**Review by dinagraves — APPROVED** [review]
Thank you! Found one more spot where the table needs to be renamed.

**Pull request merged**